### PR TITLE
replace react-syntax-highlighter with shiki

### PR DESCRIPTION
**Pull request by gsteenkamp89**
closes UMA-2783

This syntax highlighter runs only on the server. It generates markup and style tokens for both light and dark mode. Our `CodeBlock` component is now a server component, so shiki doesn't get added to the bundle at all.

test article https://uma-blog-git-feat-shiki-uma.vercel.app/articles/how-to-use-the-graph-to-query-umas-optimistic-oracle

### CODE REVIEW

File: app/layout.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ export default function RootLayout({
         <body className="min-h-dvh h-full items-center flex flex-col bg-background text-text w-screen">
           <Providers>
             <Header />
-            <main className="w-full m-x-auto items-center flex flex-col pb-8 h-full min-h-[calc(100dvh-var(--header-height)-var(--page-padding-bottom))]">
+            <main className="w-full overflow-x-clip m-x-auto items-center flex flex-col pb-8 h-full min-h-[calc(100dvh-var(--header-height)-var(--page-padding-bottom))]">
```
Comment (james-a-morris) [inline app/layout.tsx:32]: Is this change in scope?

**Review by james-a-morris — APPROVED** [review]
one nit. I can't weigh in on the pnpm lockfile.

File: app/layout.tsx (hunk 1 above)
Comment (gsteenkamp89) [reply app/layout.tsx:32]: actually has nothing to do with this task but it does fix an issue I noticed

Comment (gsteenkamp89) [reply app/layout.tsx:32]: yeah the lockfile version's been bumped automatically so the diff is kinda huge.
